### DISCUSSION TITLE
Bringing back custom indentation

### DISCUSSION THREAD

**Discussion by Fobiwan** [intersystems-community/vscode-objectscript#1297] [Ideas]
This isn't necessarily a Intersystems exclusive topic, but a feature I really liked about the studio, was the custom indentation you were able to setup.
Espacially for people who have to deal with Legacy Mumps Code and the "." indentation notation.
When you have more complex loops you sometimes go down 4-5 levels. So writing 5 dots for every line is very tedious...

I searched for a general VS Code Extension for a solution but wasn't able to find anything.
If anyone has something like this, or this would be considered for implementation I would be very happy. 

Kind Regards
Fabio

**Comment by isc-bsaviano** [comment]
@Fobiwan Can you please elaborate on what you mean by "custom indentation"? A screenshot from Studio might help. If you're talking about keyboard shortcuts, [our documentation](https://docs.intersystems.com/components/csp/docbook/DocBook.UI.Page.cls?KEY=GVSCO_fromstudio#GVSCO_fromstudio_shortcuts) has a mapping of Studio ones to their equivalents in VS Code.

**Comment by Fobiwan** [comment]
![image](https://github.com/intersystems-community/vscode-objectscript/assets/64836841/536dee2d-4cef-43e2-9cb5-a654d8bbc05f)
Sure thing, it's german but I hope you can see what I mean. 
The custom indentation adds every symbol defined at the start of a new line, if the previous line had those symbols in the front

**Reply by Fobiwan** [reply]
VS Code does this only for spaces and tabs as far as I'm aware but not for any other characters

**Reply by isc-bsaviano** [reply]
Thanks for the screenshot. We have support for auto-adding comment markers when you press "Enter" on a line with a comment (see #761) but we don't have this for legacy dot syntax

**Reply by Fobiwan** [reply]
Yes thanks I have seen that thread. Just throwing it out there because I assumed the logic is similar and might be easy to make it parameter. 
I don't expect this to be priority, but maybe some other people will come across this and see this thread.
Thank you for the information

**Reply by isc-bsaviano** [reply]
@Fobiwan I can improve the auto-comment feature (the current implementation seems to only work if there's a single space for indentation). Adding support for legacy dots may not be possible because the text you add isn't dynamically generated based on the previous line's prefix.

EDIT: I improved the auto-commenting feature with #1298
